### ISSUE TITLE
CoreOS + Azure + Weave Getting Started Guide Doesn't Work

### ISSUE THREAD

**Issue by colemickens** [GoogleCloudPlatform/kubernetes#8159]
The CoreOS/Azure cluster I brought up six days ago was affected by  #3853 with etcd unresponsive, so I tore down the cluster and created a new one.

Something has changed. `install-weave.service` fails on the master AND minions. It requires you to restart the `install-weave.service` along with other units that have failed or failed a condition and thus weren't started. (This includes apiserver, docker, controller, scheduler on the master, and docker on the minions.)

**Comment by rjnagal** [comment]
@colemickens can you get the logs for service start failure.

@errordeveloper can you help verify weave setup? Thanks!

**Comment by errordeveloper** [comment]
@colemickens, thank you for reporting this! Could you please provide the output of `systemctl status install-weave.service` from all of the nodes please?

**Comment by errordeveloper** [comment]
@colemickens what I am seeing once in a while (as I re-provision the cluster a lot while testing) is this:
```
core@kube-01 ~ $ systemctl status install-weave.service
● install-weave.service - Install Weave
   Loaded: loaded (/etc/systemd/system/install-weave.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Fri 2015-05-15 09:46:19 UTC; 32min ago
     Docs: http://weaveworks.github.io/weave/
  Process: 629 ExecStartPre=/usr/bin/curl --silent --location https://raw.github.com/errordeveloper/weave-demos/master/poseidon/weave-helper --output /opt/bin/weave-helper (code=exited, status=6)
  Process: 601 ExecStartPre=/usr/bin/curl --silent --location https://github.com/weaveworks/weave/releases/download/latest_release/weave --output /opt/bin/weave (code=exited, status=0/SUCCESS)
  Process: 579 ExecStartPre=/bin/mkdir -p /opt/bin/ (code=exited, status=0/SUCCESS)

May 15 09:46:16 kube-01 systemd[1]: Starting Install Weave...
May 15 09:46:19 kube-01 systemd[1]: install-weave.service: control process exited, code=exited status=6
May 15 09:46:19 kube-01 systemd[1]: Failed to start Install Weave.
May 15 09:46:19 kube-01 systemd[1]: Unit install-weave.service entered failed state.
May 15 09:46:19 kube-01 systemd[1]: install-weave.service failed.
```

The return code 6 implies a DNS error. What's most weird is that one script downloads okay (PID 601), while the other one fails (PID 629). I believe this is just the Azure DNS being a bit flaky at times and today I noticed more failures then before.

Unfortunately, calling `curl --retry` doesn't help and actually gets stuck forever. So I am going to add a simple wrapper to it, which I hopefully pull-request today.

My main takeaway from this is that an ultimate solution would be to avoid downloading anything and creating custom CoreOS image once in a while, but that break CoreOS updates mechanics, which are partly disabled right now. I suppose, Tectonic is aiming to solve this, but there are other question with it being non-free and how Weave integration would work on Tectonic etc; anyhow, it's not out yet.

Additionally, I had been thinking of a different kind of downloader tool, but it's a long story.

**Comment by colemickens** [comment]
Yes, this is the issue I was seeing. (It didn't repro the couple times that I tried to grab logs for this issue).

**Comment by errordeveloper** [comment]
@colemickens You can try the latest changes on [my fork](https://github.com/errordeveloper/kubernetes/).

**Comment by errordeveloper** [comment]
Perhaps it wasn't quite clear, but this has been fixed in #8323.

**Comment by rjnagal** [comment]
closing as fixed. Thanks @errordeveloper
